### PR TITLE
ddl: support drop partition for label rules

### LINKED ISSUE #26794: Support drop partition for label rules
## Development task

We should support deleting the corresponding attributes from a partition immediately once the partition is dropped. e.g., we execute the following SQL:
```
create table t1 (c int) PARTITION BY RANGE (c)
    (PARTITION p0 VALUES LESS THAN (6),
     PARTITION p1 VALUES LESS THAN (11));
alter table t1 partition p0 attributes="attr";
```
We can get the corresponding attributes for `p0` from PD.
Once we drop this partition,
```
alter table t1 drop partition p0;
```
We should not get any attributes of `p0` from PD.

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: close #26794.

This PR should be reviewed after https://github.com/pingcap/tidb/pull/26630 is merged. 

### What is changed and how it works?

What's Changed:

- Add a new function `dropLabelRules`
- Modify `onDropTablePartition` to drop the label rule immediately

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (https://github.com/pingcap/endless/pull/38/commits/4de2e3248c43a8027bd2eefb34e6e4bef8c2e707)
- [ ] No code

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by disksing — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]

File: ddl/partition.go [hunk 1]
```diff
@@ -924,6 +926,17 @@ func dropRuleBundles(d *ddlCtx, physicalTableIDs []int64) error {
...
 }
 
+func dropLabelRules(d *ddlCtx, schemaName, tableName string, partNames []string) error {
+	deleteRules := make([]string, 0, len(partNames))
+	for _, partName := range partNames {
+		deleteRules = append(deleteRules, fmt.Sprintf(label.PartitionIDFormat, label.IDPrefix, schemaName, tableName, partName))
+	}
+	// delete batch rules
+	patch := label.NewRulePatch(nil, deleteRules)
+	err := infosync.UpdateLabelRules(context.TODO(), patch)
```
Comment (xhebox) [inline ddl/partition.go:936]: Just `return infosync.UpdateLabelRules(context.TODO(), patch)`.

**Review by xhebox — COMMENTED** [review]
Refer the comment https://github.com/pingcap/tidb/pull/27102#pullrequestreview-742386183, the PR will be approved, but needs some followup.

**Review by xhebox — APPROVED** [review]

**Pull request merged**